Test failures in `master`

The last three builds in master have failed with the following:

```
--- FAIL: TestMethods (0.00s)
    --- FAIL: TestMethods/bad-methods-2.json (0.00s)
        schema_test.go:391: 
            	Error Trace:	schema_test.go:391
            	Error:      	"binding resources: error binding resource xyz:index:Foo: failed to bind methods: function xyz:index:Foo/bar for method bar is already a method" does not contain "function xyz:index:Foo/bar for method baz is already a method"
            	Test:       	TestMethods/bad-methods-2.json
```

The commit that first failed does not appear to have changed anything related though: https://github.com/pulumi/pulumi/commit/4156df31f7b91afb7deefa9110b44b39a8862217.

--- a/pkg/codegen/schema/schema.go
+++ b/pkg/codegen/schema/schema.go
@@ -1620,8 +1620,16 @@ func bindTypes(pkg *Package, complexTypes map[string]ComplexTypeSpec, loader Loa
 
 func bindMethods(resourceToken string, methods map[string]string,
 	functionTable map[string]*Function) ([]*Method, error) {
-	var result []*Method
-	for name, token := range methods {
+
+	names := make([]string, 0, len(methods))
+	for name := range methods {
+		names = append(names, name)
+	}
+	sort.Strings(names)
+
+	result := make([]*Method, 0, len(methods))
+	for _, name := range names {
+		token := methods[name]
 		function, ok := functionTable[token]
 		if !ok {
 			return nil, errors.Errorf("unknown function %s for method %s", token, name)

